engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/result/json/JsonNodeExtractor.kt

```diff
@@ -28,6 +30,34 @@ object JsonNodeExtractor {
         }
     }
 
+    /**
+     * Extract the node at the given json node path.
+     */
+    fun getNodeAt(data: JsonMap, path: JsonNodePath): JsonNode? {
+        val rootNode = JsonNode(JsonNodePath.root, data)
+        return getNodeAt(rootNode, path)
+    }
+
+    /**
+     * Extract the node at the given json node path.
+     */
+    fun getNodeAt(rootNode: JsonNode, path: JsonNodePath): JsonNode? {
+        return path.segments.fold(rootNode as JsonNode?) { currentNode, segment ->
```

Comment:
I think we should short circuit this execution once we hit null. It shouldn't impact performance too much, but no point doing extra iterations if we're always going to return null.

We can add an extension function like the following to `CollectionUtil.kt`

```kotlin
fun <T, R> Iterable<T>.foldWhileNotNull(initial: R, operation: (acc: R, T) -> R): R? {
    var accumulator = initial
    for (element in this) {
        accumulator = operation(accumulator, element) ?: return null
    }
    return accumulator
}
```

Then call `foldWhileNotNull` instead of `fold`.